lib/rubocop/cop/style/multiple_comparison.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks against comparing a variable with multiple items,
+      # where `Array.include?` could be used instead to avoid DRY errors.
+      #
+      # @example
+      #   # bad
+      #   if a == 'a' || a == 'b' || a == 'c'
+      #
+      #   # good
+      #   if ['a', 'b', 'c'].include? a
+      class MultipleComparison < Cop
+        MSG = 'Avoid comparing a variable with multiple items' \
+          'in a conditional, use include? instead.'.freeze
+
+        def on_if(node)
+          return unless nested_variable_comparison?(node.condition)
+          add_offense(node, :expression)
+        end
+
+        private
+
+        def_node_matcher :simple_double_comparison?, '(send $lvar :== $lvar)'
+        def_node_matcher :simple_lhs_comparison?, '(send $lvar :== _)'
```

Comment:
These can be combined into a single node matcher if they have the same code paths. (Looks like they do.)

```
def_node_matcher :simple_comparison?, <<-PATTERN
  {(send $lvar :== _)
   (send _ :== $lvar)}
PATTERN
```